From 1f75246f7505e4d26261267824eb5ccda8f5e09f Mon Sep 17 00:00:00 2001
From: Roman Stratiienko <roman.o.stratiienko@globallogic.com>
Date: Fri, 20 May 2022 15:06:02 +0300
Subject: [PATCH 3/4] gallium: Disable PIPE_CAP_THROTTLE on Android

To be honest I do not know how does PIPE_CAP_THROTTLING is working.
And does it make any sence on WINSYS-enabled mesa3d builds, where
external service is responsible for the frame timing.

Running simpleperf I've noticed redundant delays caused by extra fence
request at [1], which can be avoided by disabling PIPE_CAP_THROTTLE.

With this commit redundant dri_flush() call CPU time ratio to drawframe
is reduced from 1.22% to 0.15% .

[1]: https://gitlab.freedesktop.org/mesa/mesa/-/blob/2c420fcf4061b6cb8d4046c913ae10c1c8d9991d/src/gallium/frontends/dri/dri_drawable.c#L522
Signed-off-by: Roman Stratiienko <roman.o.stratiienko@globallogic.com>
---
 src/gallium/auxiliary/util/u_screen.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/gallium/auxiliary/util/u_screen.c b/src/gallium/auxiliary/util/u_screen.c
index d073e732044..b7d93a15873 100644
--- a/src/gallium/auxiliary/util/u_screen.c
+++ b/src/gallium/auxiliary/util/u_screen.c
@@ -410,7 +410,11 @@ u_pipe_screen_get_param_defaults(struct pipe_screen *pscreen,
       return 0;
 
    case PIPE_CAP_THROTTLE:
+#if defined(PIPE_OS_ANDROID)
+      return 0;
+#else
       return 1;
+#endif
 
    case PIPE_CAP_TEXTURE_SHADOW_LOD:
       return 0;
-- 
2.37.2

